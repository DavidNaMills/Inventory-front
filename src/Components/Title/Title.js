import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);