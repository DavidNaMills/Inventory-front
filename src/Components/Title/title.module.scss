@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}