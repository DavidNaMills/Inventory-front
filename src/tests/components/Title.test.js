import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import Title from '../../Components/Title/Title';

const mockMessage = 'My Test Title';


describe('<Title/> component test suite', ()=>{
    it('matches snapshot', ()=>{
        const comp = renderer.create(<Title message={mockMessage} size='h2' center/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });

    it('exists', ()=>{
        const wrapper = shallow(<Title message={mockMessage} size='h2' center/>);
        expect(wrapper.exists).toBeTruthy();
    });

    [
        {msg: 'renders a <Title/> component of size "h1"', size: 'h1', center: false, classRes: 'title_h1 '},
        {msg: 'renders a <Title/> component of size "h1" with center option set', size: 'h1', center: true, classRes: 'title_h1 title__center'},

        {msg: 'renders a <Title/> component of size "h2"', size: 'h2', center: false, classRes: 'title_h2 '},
        {msg: 'renders a <Title/> component of size "h2" with center option set', size: 'h2', center: true, classRes: 'title_h2 title__center'},
 
        {msg: 'renders a <Title/> component of size "h3"', size: 'h3', center: false, classRes: 'title_h3 '},
        {msg: 'renders a <Title/> component of size "h3" with center option set', size: 'h3', center: true, classRes: 'title_h3 title__center'},

        {msg: 'renders a <Title/> component of size "h4"', size: 'h4', center: false, classRes: 'title_h4 '},
        {msg: 'renders a <Title/> component of size "h4" with center option set', size: 'h4', center: true, classRes: 'title_h4 title__center'},
        
        {msg: 'renders a <Title/> component of size "h5"', size: 'h5', center: false, classRes: 'title_h5 '},
        {msg: 'renders a <Title/> component of size "h5" with center option set', size: 'h5', center: true, classRes: 'title_h5 title__center'},

        {msg: 'renders a <Title/> component of size "h6"', size: 'h6', center: false, classRes: 'title_h6 '},
        {msg: 'renders a <Title/> component of size "h6" with center option set', size: 'h6', center: true, classRes: 'title_h6 title__center'},
    ].forEach(x=>{
        it(x.msg, ()=>{
            const wrapper = mount(<Title message={mockMessage} size={x.size} center={x.center}/>);
            const elem = wrapper.find(x.size);
            expect(elem.length).toBe(1);
            expect(elem.props()).toHaveProperty('className', x.classRes);
            expect(elem.text()).toEqual(mockMessage);
        });
    });

    it.only('render a <Title/> component with default size "h2"', ()=>{
        const wrapper = mount(<Title message={mockMessage} center={true}/>);
        const elem = wrapper.find('h2');
        expect(elem.length).toBe(1);
        expect(elem.props()).toHaveProperty('className', 'title_h2 title__center');
        expect(elem.text()).toEqual(mockMessage);
    });
});