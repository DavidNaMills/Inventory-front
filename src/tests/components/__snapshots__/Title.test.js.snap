// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> component test suite matches snapshot 1`] = `
<div>
  <h2
    className="title_h2 title__center"
  >
    My Test Title
  </h2>
</div>
`;
